### ISSUE TITLE
WinRM - IIS Web App Deployment ERROR_SITE_DOES_NOT_EXIST

### ISSUE THREAD

**Issue by mayeager22** [Microsoft/vsts-rm-extensions#134]
I am trying to deploy to IIS, but not matter what I try for setting my site name, it does not work.

This might be related to Issue #120 

![image](https://cloud.githubusercontent.com/assets/9683192/19910116/ad7aa43c-a050-11e6-8a54-60ad8c606c58.png)

I get, but it does not tell me what it is trying to use for SiteName.

System.Exception: Error Code: ERROR_SITE_DOES_NOT_EXIST
 
2016-11-01T22:09:31.8401222Z ##[error]Deployment on one or more machines failed.
 
2016-11-01T22:09:31.8441208Z ##[error]For more info please refer to http://aka.ms/iisextnreadme
 
2016-11-01T22:09:31.9641229Z Finishing task: IISWebAppDeploy
 
2016-11-01T22:09:31.9681264Z ##[error]Task IISWebAppDeploy failed. This caused the job to fail. Look at the logs for the task for more details.

**Comment by shamnigurat** [comment]
Yep, it's the same bug as [Issue 120](https://github.com/Microsoft/vsts-rm-extensions/issues/120).

But it seems nobody hurries to fix it.

**Comment by KrishnaAdityaB** [comment]
@mayeager22 
To make sure if it is same as issue #120 can you share the steps of how you created the web package?

Can you set a Build/Release variable with name **System.Debug** & value **true** and run the Build/Release again? It will give you detailed logs and the msdeploy.exe command arguments. You can see the site name getting used.

To make sure the site exists - This task doesn't create the site. Use the **IIS - Web App Management** Task to create a site before this task. Please see the Deploy section of [this example](https://www.visualstudio.com/en-us/docs/release/examples/other-servers/net-to-vm) for reference.

**Comment by feco93** [comment]
I think this issue is same as #120.
I set the System.Debug variable to true.

Manage IIS App creates the site and application pool:
Deployment started for machine: 127.0.0.1 with port 5986.
Deployment status for machine 127.0.0.1 : Passed 
Successfully created website: ApiTest

Deploy IIS App task fails with Deployment failed on machine 127.0.0.1 with following message : System.Exception: Error Code: ERROR_SITE_DOES_NOT_EXIST message. 

I also included a parameters.xml file in the published package with the following content:
<parameters>
  <parameter name="IIS Web Application Name" defaultValue="ApiTest" tags="IisApp" />
</parameters>
And I have also tried to override IIS Web Application Name parameter.
